### PR TITLE
ReviewsAPI Implementation

### PR DESCRIPTION
**Pull request by jinyingtan**
In this PR:
- Added ReviewsAPI implementation
- Added a new `numberOfReviews` to `npos` and `donors`

### CODE REVIEW

File: utils/api/reviews.js [hunk 1]
```diff
@@ -1,23 +1,155 @@
...
+
+    const npoDoesNotExist = Object.entries(npoInfo).length === 0;
+    const donorDoesNotExist = Object.entries(donorInfo).length === 0;
+    const wishDoesNotExist = Object.entries(wishInfo).length === 0;
+
+    if (npoDoesNotExist || donorDoesNotExist || wishDoesNotExist) {
+      return false;
+    }
+
+    if (wishInfo.status !== 'pending') {
```
Comment (jamessspanggg) [inline utils/api/reviews.js:29]: should this be completed instead?

File: utils/api/reviews.js [hunk 2]
```diff
@@ -1,23 +1,155 @@
...
+
+    const npoDoesNotExist = Object.entries(npoInfo).length === 0;
+    const donorDoesNotExist = Object.entries(donorInfo).length === 0;
+    const donationDoesNotExist = Object.entries(donationInfo).length === 0;
+
+    if (npoDoesNotExist || donorDoesNotExist || donationDoesNotExist) {
+      return false;
+    }
+
+    if (donationInfo.status !== 'pending') {
```
Comment (jamessspanggg) [inline utils/api/reviews.js:86]: same goes for this

File: utils/api/reviews.js (hunk 1 above)
Comment (jinyingtan) [reply utils/api/reviews.js:29]: hmm... makes sense. I tot we would be reviewing the wish/donation before completing it.

Comment (jamessspanggg) [reply utils/api/reviews.js:29]: @kohchihao @DanKhoo what do u guys think? normally we only get to review after the transaction right

Comment (kohchihao) [reply utils/api/reviews.js:29]: it would be complete then review.

Comment (kohchihao) [reply utils/api/reviews.js:29]: This means that, the status in the DB already changed.

**Comment by kohchihao** [comment]
How to get reviews as `donor` or `npo`?

File: utils/api/reviews.js [hunk 3]
```diff
@@ -1,23 +1,181 @@
...
+      this._getDonationInfo(donationId),
+    ]);
+
+    if (donationInfo.status !== 'completed') {
+      throw new ReviewError('invalid-donation-status', 'only can review a completed donation');
+    }
+
+    const post = {
+      postId: donationId,
+      postType: 'donations',
```
Comment (jamessspanggg) [inline utils/api/reviews.js:115]: can use the constant in `constants/postType`

File: utils/api/constants.js [hunk 4]
```diff
@@ -1,2 +1,3 @@
 export const WISHES_BATCH_SIZE = 21;
 export const DONATIONS_BATCH_SIZE = 21;
+export const REVIEWS_BATCH_SIZE = 21;
```
Comment (jamessspanggg) [inline utils/api/constants.js:3]: would the batch size be too large? Perhaps 7? what do u guys think

Comment (jinyingtan) [reply utils/api/constants.js:3]: How I chose the number 21, is looking at 7 rows of 3 cards per row

Comment (jamessspanggg) [reply utils/api/constants.js:3]: I see, are we sticking with the card UI or one row for one review?

**Comment by kohchihao** [comment]
Pending biz team reply. KIV till then.

**Comment by kohchihao** [comment]
Final decision: not to include reviews into product

**Pull request closed without merging**